### PR TITLE
Updating version to 2021-06-01 for recoveryservicesbackup

### PR DESCRIPTION
**Pull request by monikajainms**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [x] new API version 
      - [x] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [x] SDK of .NET (need service team to ensure code readiness)
      - [x] SDK of Python
      - [x] SDK of Java
      - [ ] SDK of Js
      - [x] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [x] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [x] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [x] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Review by JeffreyRichter — APPROVED** [review]

File: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2018-12-20/bms.json
Comment (j5lim) [inline specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2018-12-20/bms.json:3203]: Is this only change added in 2021-06-01 as well? Is there an example file show the usage of this?

Comment (j5lim) [reply specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2018-12-20/bms.json:3203]: Next time please follow the ARM API Review Checklist that indicates how to make the very first commit for an efficient review (the second checkbox). Otherwise, it will be very hard to see what changed in the new api-version.

**Comment by j5lim** [comment]
>     "IaasVMRestoreRequest": {

TriggerCrossRegionRestore_Post.json example shows identityInfo returned in this object, but it is not modeled here.

---
Refers to: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2018-12-20/bms.json:3069 in 8f0ef99. [](commit_id = 8f0ef99820831dc10a154f374d232d969b3969ad, deletion_comment = False)

File: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2021-06-01/examples/AzureIaasVm/StopProtection.json
Comment (j5lim) [inline specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2021-06-01/examples/AzureIaasVm/StopProtection.json:9]: >2021-01-01 [](http://example.com/codeflow?start=20&length=10)

Fix the api version in all files.

**Comment by j5lim** [comment]
>     "IaasVMRestoreWithRehydrationRequest": {

Needs identityInfo in the model as the example shows.

---
Refers to: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2021-06-01/bms.json:11199 in 8f0ef99. [](commit_id = 8f0ef99820831dc10a154f374d232d969b3969ad, deletion_comment = False)

File: package-lock.json
Comment (leni-msft) [inline package-lock.json:10]: please don't check in the package-lock.json file

**Comment by filizt** [comment]
>     "version": "2021-06-01",

What is the base version of 2021-06-01?

---
Refers to: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2021-06-01/bms.json:4 in 44ea4d7. [](commit_id = 44ea4d77dc71c1e59a6f564faa7a3b168125b091, deletion_comment = False)

**Comment by filizt** [comment]
@monjain have you looked into the errors in LintDiff?

**Comment by monikajainms** [comment]
> @monjain have you looked into the errors in LintDiff?

Yes, I have added suppression for them. Please go through readme file

**Comment by monikajainms** [comment]
> What is the base version of 2021-06-01?

2021-04-01 is the base version

**Comment by filizt** [comment]
> > @monjain have you looked into the errors in LintDiff?
> 
> Yes, I have added suppression for them. Please go through readme file

It looks like the suppressions are for 2018 API version. The current LintDiff errors are for the 2021-06-01 API version.

**Comment by monikajainms** [comment]
Do not merge right now.

File: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2018-12-20/bms.json
Comment (filizt) [inline specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2018-12-20/bms.json:3194]: > "isSystemAssignedIdentity": { [](http://example.com/codeflow?start=7&length=30)

Having a bool flag is not recommended. Please use a string enum type.

File: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2018-12-20/bms.json
Comment (filizt) [inline specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2018-12-20/bms.json:3178]: >"isSystemAssignedIdentity": { [](http://example.com/codeflow?start=8&length=29)

Same as below, having a bool flag is not recommended. Please use a string enum type.

File: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2018-12-20/examples/AzureIaasVm/GetAccessToken_Get.json
Comment (filizt) [inline specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2018-12-20/examples/AzureIaasVm/GetAccessToken_Get.json:18]: > "managedIdentityArmUrl": "/subscriptions/00000000-0000-0000-0000-000000000000/resourcegroups/asmaskarRG1/providers/Microsoft.ManagedIdentity/userAssignedIdentities/asmaskartestmsi" [](http://example.com/codeflow?start=9&length=181)

You might want to rename this property to managedIdentityResourceId since the example you have is a resource Id and not a URL(with host and other information)

File: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2018-12-20/bms.json
Comment (monikajainms) [reply specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2018-12-20/bms.json:3194]: In past ARM PR review we got this suggestion from ARM team that, having a “Type” and maintaining an enum is an overhead and leads to issues. With this reasoning we were suggested to use bool for a similar property.
We already use this in existing API. For this, we prefer to keep it as is.

File: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2018-12-20/bms.json
Comment (monikajainms) [reply specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2018-12-20/bms.json:3178]: same, In past ARM PR review we got this suggestion from ARM team that, having a “Type” and maintaining an enum is an overhead and leads to issues. With this reasoning we were suggested to use bool for a similar property.
We already use this in existing API. For this, we prefer to keep it as is.

File: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2018-12-20/examples/AzureIaasVm/GetAccessToken_Get.json
Comment (monikajainms) [reply specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2018-12-20/examples/AzureIaasVm/GetAccessToken_Get.json:18]: We use it as part of URL by prefixing with ARM endpoint for all crud operations on the resource.

Comment (pilor) [reply specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2018-12-20/examples/AzureIaasVm/GetAccessToken_Get.json:18]: How its used internally is not relevant.  All resource IDs can be turned into URLs by prepending a hostname.  The string is a resource ID, not a URL.  Other properties in this object are also resource IDs and use a "resourceId" suffix.  Either change this property name to "managedIdentityId" or "managedIdentityResourceId".

Comment (monikajainms) [reply specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2018-12-20/examples/AzureIaasVm/GetAccessToken_Get.json]: done, please check

File: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2018-12-20/examples/AzureIaasVm/GetAccessToken_Get.json
Comment (pilor) [inline specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2018-12-20/examples/AzureIaasVm/GetAccessToken_Get.json:50]: These URLs are internal and not exposed to customers.  If you feel there is some scenario that requires it to be exposed please get explicit approval from the managed identity team (https://armwiki.azurewebsites.net/authorization/managed_identities/MSIOnboardingContactUs.html)

File: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2018-12-20/bms.json
Comment (jorgecotillo) [reply specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2018-12-20/bms.json:3181]: +1 This is super important otherwise we'll need to review all the changes and will be adding comments to unmodified changes.

**Pull request merged**
